integrations/aws/aws/aws_credentials.py

```diff
@@ -33,24 +120,22 @@ async def update_enabled_regions(self) -> None:
                 if region["RegionOptStatus"] == "ENABLED_BY_DEFAULT"
             ]
 
-    def is_role(self) -> bool:
-        return self.session_token is not None
-
-    async def create_session(self, region: Optional[str] = None) -> aioboto3.Session:
-        if self.is_role():
-            return aioboto3.Session(
-                self.access_key_id, self.secret_access_key, self.session_token, region
-            )
-        else:
-            return aioboto3.Session(
-                aws_access_key_id=self.access_key_id,
-                aws_secret_access_key=self.secret_access_key,
-                region_name=region,
-            )
-
-    async def create_session_for_each_region(
+    async def create_refreshable_session_for_each_region(
```

Comment:
I don't see why change the name pf this function, from an API standpoint the person using this function it will not matter if it's refreshable or not, WDYT?

Reply:
I agree